### PR TITLE
Use symmetrize.cf in reading functions

### PR DESCRIPTION
**Pull request by martin-ueding**
See #66.

### CODE REVIEW

File: R/readutils.R [hunk 1]
```diff
@@ -412,10 +395,16 @@ readbinarycf <- function(files,
     }
   }
 
-  cf <- cf_meta(nrObs = 1, Time=T, nrStypes = 1, symmetrised = symmetrise)
-  cf <- cf_orig(cf, cf = t(Re(Cf)), icf = t(Cf))
+  ret <- cf_meta(nrObs = 1, Time=T, nrStypes = 1, symmetrised = symmetrise)
+  ret <- cf_orig(ret, cf = t(Re(Cf)), icf = t(Cf))
```
Comment (kostrzewa) [inline R/readutils.R]: ```Im``` missing

**Comment by martin-ueding** [comment]
It was missing in two cases, `readtextcf` seemed to be correct. Is this okay to merge now?

File: R/readutils.R [hunk 2]
```diff
@@ -470,7 +451,11 @@ readbinarysamples <- function(files, T=48, nosamples=2, endian="little",
   ret <- list()
   for (i in 1:nosamples) {
     ret[[i]] <- cf_meta(nrObs = 1, Time=T, nrStypes = 1, symmetrised = symmetrise)
```
Comment (kostrzewa) [inline R/readutils.R]: this is actually currently broken because ```symmetrise``` does not exist...

**Comment by kostrzewa** [comment]
The whole set of reading routines needs some work because some things have become inconsistent. Most notably how smearing and observables are treated...

File: R/readutils.R [hunk 3]
```diff
@@ -458,9 +442,6 @@ readbinarysamples <- function(files, T=48, nosamples=2, endian="little",
         } else {
           tmp2 <- apply(X=tmp[,1:i],MARGIN=1,FUN=mean)
         }
-        # average over +- t
-        tmp2[i1] <- 0.5 * ( tmp2[i1] + sign * tmp2[i2] )
-        Cf[[i]] <- cbind(Cf[[i]],tmp2[1:(T/2+1)])
```
Comment (kostrzewa) [inline R/readutils.R]: you broke functionality here

File: R/readutils.R (hunk 2 above)
Comment (kostrzewa) [reply R/readutils.R]: ping?

**Comment by kostrzewa** [comment]
@martin-ueding is this abandoned?

**Comment by martin-ueding** [comment]
Sorry for not getting back to this. I have just looked at it again and realize that indeed this breaks things. In the constructor I now use `symmetrize = TRUE`, so that should be fine. And for the actual data I now append the full correlation function and then symmetrize them at the end with the new function.

Is there some sample data where I could test this with?

**Comment by kostrzewa** [comment]
There's also still the problem with the ```readbinarysamples``` function.

I've added example data to the package in https://github.com/hiskp-lqcd/hadron_example_data

**Comment by martin-ueding** [comment]
I have just looked into this sample data repository but I only found text files containing data. If the error is in the `readbinarysamples` function, I would need binary data, right?

**Comment by martin-ueding** [comment]
I have tested the `readbinarysamples` function with the test data, the shape of the output looks sensible now.

**Pull request merged**